sql/sql_delete.cc

```diff
@@ -1182,25 +1237,46 @@ multi_delete::initialize_tables(JOIN *join)
       tbl->prepare_triggers_for_delete_stmt_or_event();
       tbl->prepare_for_position();
       tbl->file->prepare_for_modify(tbl->versioned(VERS_TIMESTAMP), true);
-    }
-    else if ((tab->type != JT_SYSTEM && tab->type != JT_CONST) &&
-             walk == delete_tables)
-    {
-      /*
-        We are not deleting from the table we are scanning. In this
-        case send_data() shouldn't delete any rows a we may touch
-        the rows in the deleted table many times
-      */
-      delete_while_scanning= false;
+
+      // maybe don't do this if join->need_tmp is false, dunno
+      List<Item> temp_fields;
+      tbl->prepare_for_position();
+      join->map2table[tbl->tablenr]->keep_current_rowid= true;
+      Item_temptable_rowid *item=
+        new (thd->mem_root) Item_temptable_rowid(tbl);
+      if (!item)
+         DBUG_RETURN(1);
+      item->fix_fields(thd, 0);
+      if (temp_fields.push_back(item, thd->mem_root))
+        DBUG_RETURN(1);
+      /* Make an unique key over the first field to avoid duplicated updates */
+      ORDER group;
+      bzero((char*) &group, sizeof(group));
+      group.direction= ORDER::ORDER_ASC;
+      group.item= (Item**) temp_fields.head_ref();
+      TMP_TABLE_PARAM *tmp_param;
+      prior->shared = index;
+      tmp_param= &tmp_table_param[prior->shared];
+      tmp_param->init();
+      tmp_param->tmp_name="update";
+      tmp_param->field_count= temp_fields.elements;
+      tmp_param->func_count=  temp_fields.elements - 1;
+      calc_group_buffer(tmp_param, &group);
+      /* small table, ignore @@big_tables */
```

Comment:
This looks cryptic for a casual reader. Is this "ignore @@big_tables" introduced by this patch? If yes  please elaborate about the reason...

Reply:
It seems that this is no longer needed as of 11.4.  It was deprecated in 10.5 according to [this KB page](https://mariadb.com/kb/en/server-system-variables/).  Our KB has [another page](https://mariadb.com/docs/server/ref/mdb/system-variables/big_tables/) which says that this is an "old variable, which if set to 1, allows large result sets by saving all temporary sets to disk, avoiding 'table full' errors. No longer needed, as the server now handles this automatically."  I saw this when I was reading the `multi_update` code and preserved it in my own implementation for `multi_delete` as it seems reasonable, but allowing the system to decide automatically is better.